src/agent/src/agent.cpp

```diff
@@ -43,6 +59,52 @@ void Agent::Run()
         [this]([[maybe_unused]] const std::string& response)
         { PopMessagesFromQueue(m_messageQueue, MessageType::STATELESS); }));
 
+    m_taskManager.EnqueueTask(m_commandHandler.ProcessCommandsFromQueue<command_store::Command>(
+        [this]() -> std::optional<command_store::Command>
```

Comment:
Maybe we can create a function similar to `GetMessagesFromQueue` here instead of this lambda.

Reply:
Done